dataans/src-tauri/src/dialog.rs

```diff
@@ -0,0 +1,27 @@
+use futures::channel::oneshot;
+use std::path::PathBuf;
+use tauri::{command, AppHandle};
+use tauri_plugin_dialog::{DialogExt, FilePath};
+
+#[command]
+pub async fn open_file_dialog(app: AppHandle) -> Result<Option<PathBuf>, String> {
```

Comment:
```suggestion
pub async fn select_file(app: AppHandle) -> Commandresult<PathBuf> {
```

1. use appropriate naming
2. Why didn't you use the `CommandResult`?